byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+if(document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', loadButtons);
+} else {
+  loadButtons();
+}
+
+let interestsChosen = [];
+
+/** 
+ * Track the interests chosen in an array according to their current
+ * status. Toggle their status afterwards.
+ * @param {Element elem} tracks the current button element chosen.
+ */
+function toggleStatus(elem) {
```

Comment:
Here's how to do what I suggested:

instead of taking just a button element, you can take both the button to toggle and the name/id of the interest that the button is for:

`function toggleStatus(place, elem)`

In the createButtonForPlace function, when you rig the event-handler, instead of just passing the button element, you can also pass along the `place` that that button is associated with:

`button.addEventListener('click', () => toggleStatus(place, button));`

then inside of this function, you're able to compare `place` with the contents of the list, and explicitly update the classes based on it (note that since each interest should only be selected once, you may want to use a [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) rather than a regular js Array)

```js
// note: this example assumes that this is using "set"
if (interestsChosen.has(place)) {
   interestsChosen.delete(place);
   elem.classList.remove('active');
} else {
  interestsChosen.add(place);
  elem.classList.add('active');
}
```

Reply:
I see, I interpreted it as adding onto the array then looping through that array and through each button to change the status. I wanted to avoid it since that'd take up a lot of runtime per click, which should be pretty quick. But I see now that you can check to add/remove right away rather than afterwards. Also, I didn't know sets were a thing in JS since there don't seem to be "real" types in JS, so thanks. Fixed